### PR TITLE
Add setting enable_optimize_query_tree_logical_expression

### PR DESCRIPTION
**Pull request by vdimir**
### Changelog category (leave one):

- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
* Added the setting `enable_optimize_query_tree_logical_expression` to disable the optimization of the logical optimizer on top of the query tree (enabled by default).

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: tests/clickhouse-test [hunk 1]
```diff
@@ -897,6 +897,8 @@ class MergeTreeSettingsRandomizer:
         "cache_populated_by_fetch": lambda: random.randint(0, 1),
         "concurrent_part_removal_threshold": threshold_generator(0.2, 0.3, 0, 100),
         "old_parts_lifetime": threshold_generator(0.2, 0.3, 10, 8 * 60),
+        # "enable_optimize_query_tree_logical_expression": lambda: random.randint(0, 1),
+        "enable_optimize_query_tree_logical_expression": int,
```
Comment (vdimir) [inline tests/clickhouse-test:901]: Once CI is passed with `enable_optimize_query_tree_logical_expression = 0` and all tests are adjusted, I will set it to random ( maybe only for some builds/runs ? )

```suggestion
        "enable_optimize_query_tree_logical_expression": lambda: random.randint(0, 1),
```

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit f3dc3296f0b63539b11ae616be4bab3a747099ae with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/68056/f3dc3296f0b63539b11ae616be4bab3a747099ae/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/68056/f3dc3296f0b63539b11ae616be4bab3a747099ae/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by jkartseva** [comment]
Tests need to be adjusted.

**Review by jkartseva — APPROVED** [review]
Looks good.

**Comment by jkartseva** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/68056/e9255e614bf0b6eba0b35ece1fffbbd740523b39/integration_tests__aarch64__[6_6].html -- has failed

**Comment by vdimir** [comment]
> [s3.amazonaws.com/clickhouse-test-reports/68056/e9255e614bf0b6eba0b35ece1fffbbd740523b39/integration_tests__aarch64__[6_6].html](https://s3.amazonaws.com/clickhouse-test-reports/68056/e9255e614bf0b6eba0b35ece1fffbbd740523b39/integration_tests__aarch64__%5B6_6%5D.html) -- has failed

Seems like this issue https://github.com/ClickHouse/ClickHouse/issues/68425

**Comment by vdimir** [comment]
`SettingsChangesHistory.cpp` conflict very often...

**Pull request closed without merging**